src/shared/tpm2-util.c

```diff
@@ -1393,6 +1395,20 @@ static void hash_pin(const char *pin, size_t len, TPM2B_AUTH *auth) {
         sha256_finish_ctx(&hash, auth->buffer);
 }
 
+static bool tpm2_is_encryption_session(Tpm2Context *c, const Tpm2Handle *session) {
+        TPMA_SESSION flags = 0;
+        TSS2_RC rc;
+
+        assert(c);
+        assert(session);
+
+        rc = sym_Esys_TRSess_GetAttributes(c->esys_context, session->esys_handle, &flags);
+        if (rc != TSS2_RC_SUCCESS)
+                return false;
+
+        return flags & TPMA_SESSION_DECRYPT && flags & TPMA_SESSION_ENCRYPT;
```

Comment:
`(flags & TPMA_SESSION_DECRYPT) && (flags & TPMA_SESSION_ENCRYPT)` to make it more readable

Reply:
ack